adam_core/coordinates/covariances.py

```diff
@@ -78,12 +78,35 @@ def to_matrix(self) -> np.ndarray:
             Covariance matrices for N coordinates in 6 dimensions.
         """
         # return self.values.combine_chunks().to_numpy_ndarray()
-        cov = np.stack(self.values.to_numpy(zero_copy_only=False))
-        if np.all(cov == None):  # noqa: E711
+        values = self.values.to_numpy(zero_copy_only=False)
+
+        # If all covariance matrices are None, then return a covariances
+        # filled with NaNs.
+        if np.all(values == None):  # noqa: E711
```

Comment:
Arrow arrays keep track of the number of null elements they have in a property called `null_count`. That could be quite a lot faster here - it would just look up two integers instead of converting everything to Python objects and comparing item-by-item:

```py
if self.values.null_count == len(self.values):
    return np.full((len(self.values), 6, 6), np.nan)
```